tutorminio/templates/minio/tasks/minio/init.sh

```diff
@@ -1,9 +1,36 @@
+# Verify that `MINIO_GCS_APPLICATION_CREDENTIALS` and `MINIO_GCS_APPLICATION_ID` are defined if `MINIO_GATEWAY == "gcs"`
+{% if MINIO_GATEWAY == "gcs" %}
+  {% if not MINIO_GCS_APPLICATION_CREDENTIALS %}
+    echo "⚠️  MINIO_GCS_APPLICATION_CREDENTIALS is not defined. It is required to when MINIO_GATEWAY is set to gcs."
+    exit 1
+  {% endif %}
+  {% if not MINIO_GCS_APPLICATION_ID %}
+    echo "⚠️  MINIO_GCS_APPLICATION_ID is not defined. It is required to when MINIO_GATEWAY is set to gcs."
+    exit 1
+  {% endif %}
+{% endif %}
+
 mc config host add minio http://minio:9000 {{ OPENEDX_AWS_ACCESS_KEY }} {{ OPENEDX_AWS_SECRET_ACCESS_KEY }} --api s3v4
-mc mb --ignore-existing minio/{{ MINIO_BUCKET_NAME }} minio/{{ MINIO_FILE_UPLOAD_BUCKET_NAME }} minio/{{ MINIO_VIDEO_UPLOAD_BUCKET_NAME }}
 
-{% if MINIO_GATEWAY != "azure" %}
+BUCKETS=(
+    "{{ MINIO_BUCKET_NAME }}"
+    "{{ MINIO_FILE_UPLOAD_BUCKET_NAME }}"
+    "{{ MINIO_VIDEO_UPLOAD_BUCKET_NAME }}"
+)
+
+for BUCKET in "${BUCKETS[@]}"; do
+    if mc ls minio/ | grep -q "${BUCKET}/$"; then
+        echo "Bucket ${BUCKET} already exists!"
+    else
+        mc mb minio/${BUCKET}
+    fi
+done
+
```

Comment:
I'm curious about this `mc` command (I take is the midnight commander ?)

Reply:
it's actually `minio mc` where `mc` stands for `MinIO Client`. But we don't use `minio` and we use `mc` directly because we run the command inside MinIO shell

Reply:
Exactly! It was funny to think it was the Mindnight commander